app/controllers/ops_controller/settings/ldap.rb

```diff
@@ -328,6 +326,21 @@ def ls_delete
 
   private
 
+  def checked_or_params_id(klass)
+    objs = find_checked_items
+    obj = objs.blank? && params[:id] ? params[:id] : objs[0]
+    obj = from_cid(obj) if obj.present?
+    obj
+  end
+
+  def find_or_new(klass, ids)
```

Comment:
Does not make sense to call the method if you don't expect to have an id. Then the decision on what to do would not be to be done in the method.